applications/lib_app/rsa_util/rsa_util.cpp

```diff
@@ -252,3 +252,42 @@ bool openssl_gen_dh(const char* dh_key, int bitsize)
     return true;
 #endif
 }
+
+bool get_server_name(const char *server_crt_path, char *server_name)
+{
+    char temp[256];
+
+    FILE *f = fopen(server_crt_path, "r");
+    if (f == NULL) {
+        return false;
+    }
+
+    while (fgets(temp, sizeof(temp), f) > 0) {
+        sscanf(temp, "\t\tSubject: %[^\n]", server_name);
+    }
+
+    fclose(f);
+
+    return true;
+}
+
+bool get_ca_name(const char *ca_crt_path, char *ca_name)
+{
+    char temp[256];
+    char command_line[256];
+
+    snprintf(command_line, sizeof(command_line), "openssl x509 -in %s -noout -text", ca_crt_path);
+
+    FILE *f = popen(command_line, "r");
+    if (f == NULL) {
+        return false;
+    }
+
+    while (fgets(temp, sizeof(temp), f) > 0) {
+        sscanf(temp, "\t\tSubject: %[^\n]", ca_name);
```

Comment:
What happen if ca_name isn't long enough ?

Reply:
Tri should use a command to get all subject of ca: openssl x509 -noout -subject -in ca.crt. There are result for this command: subject= /CN=CA/ST=NC/C=VN/emailAddress=rootCA@gmail.com/O=Example/OU=IT Department.